app/src/main/java/org/simple/clinic/home/overdue/OverdueAppointmentRow.kt

```diff
@@ -103,6 +106,12 @@ data class OverdueAppointmentRow(
 
     holder.patientNameTextView.text = context.getString(R.string.overdue_list_item_name_age, name, age.toString())
     holder.patientNameTextView.setCompoundDrawableStart(gender.displayIconRes)
+    holder.patientAddressTextView.text = if (patientAddress.streetAddress != null && patientAddress.colonyOrVillage != null) {
+      "${patientAddress.streetAddress}, ${patientAddress.colonyOrVillage}"
+    } else {
+      patientAddress.streetAddress ?: patientAddress.colonyOrVillage
+      ?: "${patientAddress.district}, ${patientAddress.state}"
+    }
```

Comment:
Can we replace this multiple if-else and elvis operators with a single level `when` clause?

Reply:
Do you mean something like this?
```kotlin
holder.patientAddressTextView.text = when {
  patientAddress.streetAddress != null && patientAddress.colonyOrVillage != null -> {
    "${patientAddress.streetAddress}, ${patientAddress.colonyOrVillage}"
  }
  patientAddress.streetAddress != null -> {
    patientAddress.streetAddress
  }
  patientAddress.colonyOrVillage != null -> {
    patientAddress.colonyOrVillage
  }
  else -> {
    "${patientAddress.district}, ${patientAddress.state}"
  }
}
```

Reply:
Yes, it seems more readable. Most of these can also have the braces removed.